packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -82,50 +119,124 @@ const shouldWriteExecutionToRDS = async (cumulusMessage, knex) => {
-  const executionArn = getMessageExecutionArn(cumulusMessage);
-
-  const isRDSWriteEnabled = await shouldWriteExecutionToRDS(cumulusMessage, knex);
+const saveExecutionViaTransaction = async ({ cumulusMessage, trx }) =>
+  trx(tableNames.executions)
+    .insert({
+      arn: getMessageExecutionArn(cumulusMessage),
+      cumulus_version: getMessageCumulusVersion(cumulusMessage),
+    });
 
-  if (!isRDSWriteEnabled) {
+const saveExecution = async ({
+  cumulusMessage,
+  knex,
+  executionModel = new Execution(),
+}) =>
+  knex.transaction(async (trx) => {
+    await saveExecutionViaTransaction({ cumulusMessage, trx });
     return executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
-  }
+  });
 
-  try {
-    return await knex.transaction(async (trx) => {
-      await trx(tableNames.executions)
-        .insert({
-          arn: executionArn,
-          cumulus_version: getMessageCumulusVersion(cumulusMessage),
-        });
-      return executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
+const savePdrViaTransaction = async ({
+  cumulusMessage,
+  collection,
+  provider,
+  trx,
+}) =>
+  trx(tableNames.pdrs)
+    .insert({
+      name: getMessagePdrName(cumulusMessage),
+      status: getWorkflowStatus(cumulusMessage),
+      collectionCumulusId: collection.cumulusId,
+      providerCumulusId: provider.cumulusId,
     });
-  } catch (error) {
-    log.error(`Failed to write execution records for ${executionArn}`, error);
-    throw error;
+
+const savePdr = async ({
+  cumulusMessage,
+  collection,
+  provider,
+  knex,
+  pdrModel = new Pdr(),
+}) => {
+  // If there is no PDR in the message, then there's nothing to do here, which is fine
+  if (!messageHasPdr(cumulusMessage)) {
+    return true;
+  }
+  if (!isRecordDefined(collection)) {
+    throw new Error(`Collection reference is required for a PDR, got ${collection}`);
   }
+  if (!isRecordDefined(provider)) {
+    throw new Error(`Provider reference is required for a PDR, got ${provider}`);
+  }
+  return knex.transaction(async (trx) => {
+    await savePdrViaTransaction({ cumulusMessage, collection, provider, trx });
+    return pdrModel.storePdrFromCumulusMessage(cumulusMessage);
+  });
 };
 
-const savePdrToDb = async (cumulusMessage) => {
-  const pdrModel = new Pdr();
+const saveGranulesToDb = async (cumulusMessage) => {
+  const granuleModel = new Granule();
+
   try {
-    await pdrModel.storePdrFromCumulusMessage(cumulusMessage);
+    await granuleModel.storeGranulesFromCumulusMessage(cumulusMessage);
   } catch (error) {
     const executionArn = getMessageExecutionArn(cumulusMessage);
-    log.fatal(`Failed to create/update PDR database record for execution ${executionArn}: ${error.message}`);
+    log.fatal(`Failed to create/update granule records for execution ${executionArn}: ${error.message}`);
     throw error;
   }
 };
 
-const saveGranulesToDb = async (cumulusMessage) => {
+const saveRecordsToDynamoDb = async (cumulusMessage) => {
```

Comment:
Noting for my own sanity - this is a re-implementation of the Dynamo logic originally present if isExecutionRDSWriteEnabled is set, instead of implementing something that makes the various 'save' calls conditional.   

This approach seems fine.